lld/test/ELF/aarch64-reloc-implicit-addend.test

```diff
@@ -1,97 +1,318 @@
+# RUN: llvm-objdump -s %t/a.out | FileCheck %s --check-prefix=DATA
+# RUN: llvm-objdump -d %t/a.out | FileCheck %s --check-prefix=CODE
+
+#--- Inputs/symbols.s
+
+// Source file containing the values of target symbols for the relocations. If
+// we don't keep these in their own file, then llvm-mc is clever enough to
+// resolve some of the relocations during assembly, even though they're written
+// as explicit .reloc directives. But we want the relocations to be present in
+// the object file, so that yaml2obj can change their type and we can test
+// lld's handling of the result. So we ensure that llvm-mc can't see both the
+// .reloc and the target symbol value at the same time.
+
+.globl abs16
+.globl abs32
+.globl abs64
+.globl big64
+.globl pcrel
+.globl data
+.globl branchtarget
+.globl calltarget
+
+.equ abs16, 0x9999
+.equ data, 0x100000
+.equ branchtarget, 0x200100
+.equ calltarget, 0x02000100
+.equ pcrel, 0x245678
+.equ abs32, 0x88888888
+.equ abs64, 0x7777777777777777
+.equ big64, 0x77ffffffffffff77
+
+#--- Inputs/relocs.s
+
+// Source file containing the test instructions and their relocations, with the
+// FileCheck comments interleaved.
+
+// DATA: Contents of section .data:
+.data
+
+// First test absolute data relocations. For each one I show the expected
+// value in a comment, and then expect a line in llvm-objdump -s containing
+// all the values together.
+
+        // 0x7777777777777777 + 0x1234567887654321 = 0x89abcdeffedcba98
+        .reloc ., R_AARCH64_ABS64, abs64
+        .xword 0x1234567887654321
+
+        // 0x88888888 + 0x12344321 = 0x9abccba9
+        .reloc ., R_AARCH64_ABS32, abs32
+        .word 0x12344321
+
+        // 0x9999 + 0x1234 = 0xabcd
+        .reloc ., R_AARCH64_ABS16, abs16
+        .hword 0x1234
+
+        // DATA-NEXT:  100000 98badcfe efcdab89 a9cbbc9a cdab
+
+        .balign 16
+
+// Test relative data relocs, each subtracting the address of the relocated
+// word.
+
+        // 0x100000 + 0x1234567887654321 - 0x100010 = 0x1234567887654311
+        .reloc ., R_AARCH64_PREL64, data
+        .xword 0x1234567887654321
+
+        // 0x100000 + 0x12344321 - 0x100018 = 0x12344309
+        .reloc ., R_AARCH64_PREL32, data
+        .word 0x12344321
+
+        // 0x100000 + 0x1234 - 0x10001c = 0x1218
+        .reloc ., R_AARCH64_PREL16, data
+        .hword 0x1234
+
+        // DATA-NEXT:  100010 11436587 78563412 09433412 1812
+
+// CODE: 0000000000200000 <_start>:
+.text
+.globl _start
+_start:
+
+// Full set of 4 instructions loading the constant 'abs64' and adding 0x1234 to
+// it.
+
+// Expected constant is 0x7777777777777777 + 0x1234 = 0x77777777777789ab
+
+        .reloc ., R_AARCH64_MOVW_UABS_G0_NC, abs64
+        movz x0, #0x1234
+        // CODE-NEXT:  200000: d2913560      mov     x0, #0x89ab
+        .reloc ., R_AARCH64_MOVW_UABS_G1_NC, abs64
+        movk x0, #0x1234, lsl #16
+        // CODE-NEXT:  200004: f2aeeee0      movk    x0, #0x7777, lsl #16
```

Comment:
We can remove the leading addresses so that changing the address would not require updates to every CHECK line.

In other tests, we keep some addresses when the instruction is referenced. I think that doesn't apply to this test.